pkg/registry/images.go

```diff
@@ -0,0 +1,245 @@
+package registry
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/containers/image/v5/docker"
+	imagetypes "github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+	dockerregistry "github.com/replicatedhq/kots/pkg/docker/registry"
+	"github.com/replicatedhq/kots/pkg/image"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/registry/types"
+	"github.com/replicatedhq/kots/pkg/store"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+var deleteImagesTaskID = "delete-images"
+
+func DeleteUnusedImages(ctx context.Context, registry types.RegistrySettings, usedImages []string) (finalError error) {
+	if registry.Hostname == "" {
+		return nil
+	}
+
+	currentStatus, _, err := store.GetStore().GetTaskStatus(deleteImagesTaskID)
+	if err != nil {
+		return errors.Wrap(err, "failed to get task status")
+	}
+
+	if currentStatus == "running" {
+		logger.Debug("update-download is already running, not starting a new one")
+		return nil
+	}
+
+	if err := store.GetStore().SetTaskStatus(deleteImagesTaskID, "Searching registry...", "running"); err != nil {
+		return errors.Wrap(err, "failed to set task status")
+	}
+
+	finishedChan := make(chan error)
+	defer close(finishedChan)
+
+	startDeleteImagesTaskMonitor(finishedChan)
+	defer func() {
+		finishedChan <- finalError
+	}()
+
+	sysCtx := &imagetypes.SystemContext{
+		DockerInsecureSkipTLSVerify: imagetypes.OptionalBoolTrue,
+		DockerDisableV1Ping:         true,
+	}
+	if registry.Username != "" && registry.Password != "" {
+		sysCtx.DockerAuthConfig = &imagetypes.DockerAuthConfig{
+			Username: registry.Username,
+			Password: registry.Password,
+		}
+	}
+
+	searchResult, err := docker.SearchRegistry(ctx, sysCtx, registry.Hostname, "", math.MaxInt32)
+	if err != nil {
+		return errors.Wrap(err, "failed to seacrh registry")
+	}
+
+	digestsInRegistry := map[string]string{}
+	for _, r := range searchResult {
+		imageName := path.Join(registry.Hostname, r.Name)
+		ref, err := docker.ParseReference(fmt.Sprintf("//%s", imageName))
+		if err != nil {
+			logger.Infof("failed to parse registry image ref %q: %v", imageName, err)
```

Comment:
all/most `.Infof` in this function should be `.Errorf` IMO
```suggestion
			logger.Errorf("failed to parse registry image ref %q: %v", imageName, err)
```